Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Specific/SwapChainPass.cpp

```diff
@@ -102,15 +103,40 @@ namespace AZ
             params.m_scissorState = m_scissorState;
             params.m_viewportState = m_viewportState;
 
-            if(m_windowContext->GetSwapChain() == nullptr || m_windowContext->GetSwapChain()->GetImageCount() == 0)
+            if (m_windowContext->GetSwapChain(m_swapchainMode) == nullptr ||
+                m_windowContext->GetSwapChain(m_swapchainMode)->GetImageCount() == 0)
             {
                 return;
             }
 
             RHI::FrameGraphAttachmentInterface attachmentDatabase = params.m_frameGraphBuilder->GetAttachmentDatabase();
 
+            AZ::RPI::RPISystemInterface* rpiSystem = AZ::RPI::RPISystemInterface::Get();
+            if (rpiSystem->GetXRSystem())
+            {
+                //Acquire the XR swapchain images from the Xr back end
+                switch (m_swapchainMode)
+                {
+                    case WindowContext::SwapChainMode::XrLeft:
+                    {
+                        rpiSystem->GetXRSystem()->AcquireSwapChainImage(0);
```

Comment:
Maybe cast the `XrLeft`/`XrRight` enums to `u32` and use those as the indices here.

Reply:
I cant do that here as XrLeft = 1 and XrRight is 2

Reply:
Hm, I thought I saw them as 0 and 1.